packages/foundation-models/src/azure-openai/azure-openai-chat-completion-response.test.ts

```diff
@@ -14,9 +15,42 @@ describe('OpenAI chat completion response', () => {
     headers: {},
     request: {}
   };
-  const response = new AzureOpenAiChatCompletionResponse(rawResponse);
+  const azureOpenAiChatResponse = new AzureOpenAiChatCompletionResponse(
+    rawResponse
+  );
+
+  it('should return the chat completion response', () => {
+    expect(azureOpenAiChatResponse.data).toStrictEqual(mockResponse);
+  });
+
+  it('should return raw response', () => {
+    expect(azureOpenAiChatResponse.rawResponse).toBe(rawResponse);
+  });
+
+  it('should get token usage', () => {
+    expect(azureOpenAiChatResponse.getTokenUsage()).toMatchObject({
+      completion_tokens: expect.any(Number),
+      prompt_tokens: expect.any(Number),
+      total_tokens: expect.any(Number)
+    });
+  });
+
+  it('should return default choice index with convenience functions', () => {
+    expect(azureOpenAiChatResponse.getFinishReason()).toBe('stop');
+    expect(azureOpenAiChatResponse.getContent()).toBe(
+      "Hello! I'm just a computer program, so I don't have feelings, but thanks for asking. How can I assist you today?"
+    );
+  });
 
-  it('should return the completion response', () => {
-    expect(response.data).toStrictEqual(mockResponse);
+  it('should return undefined when convenience function is called with incorrect index', () => {
+    const logger = createLogger({
+      package: 'foundation-models',
+      messageContext: 'azure-openai-chat-completion-response'
+    });
+    const errorSpy = jest.spyOn(logger, 'error');
```

Comment:
[q] Can we spy on the private method `logInvalidDataIndex` directly?

```ts
const spy = jest.spyOn(azureOpenAiChatResponse as any, 'logInvalidDataIndex');
expect(spy).toHaveBeenCalledWith(1);
expect(spy).toHaveBeenCalledTimes(1);
```

Reply:
While possible,  I don't know if hacking our way around typescript is the best practice here.
On the top of my mind, I don't see a strong reason opposing this approach though.

Reply:
The thing is this function will always be called, but doesn't mean the logger will also be. I want to test that the logger.error is called.